### PR TITLE
docs: add instructions for e5 serve target

### PR DESCRIPTION
**Pull request by jbogarthyde**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [x] Documentation content changes
- [ ] angular.io application / infrastructure changes
- [ ] Other... Please describe:

## What is the current behavior?
Issue Number:  https://github.com/angular/angular/issues/31071

## What is the new behavior?
- Differential load mentioned on Browser Support page
- Instructions added to Diff Load section of Deployment page for running serve/test on e5 platform

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

**Comment by alastair-todd** [comment]
@jbogarthyde as far as I understand it, the inclusion of the necessary "Mandatory" es6 polyfills section for IE and friends is no longer a manual step (like in previous versions when we had to add multiple core-js imports). remember this: https://unpkg.com/ccbl-gfx-angular-test@0.0.6/src/polyfills.ts

I still think that section requires simplification because how else will long term users interpret what its saying?

I can't find where i read it, but its effectively redundant because its replaced by differential loading (as far as I understand).

**Comment by alastair-todd** [comment]
@jbogarthyde the more i look at the doc the more I dislike it (sorry).
Can you please just add a step by step instruction to add build support for IE11 in the recommended way? This is far more important to the majority than saving a few Kb on the app payload assets.
Also another breaking change that occurred recently was that autoprefixer turned off css grid support by default. There's zero instruction on this in the angular docs (that autoprefixer is even involved).

It effectively took me 1 whole day to decide not to use css grid in our new project (even though i eventually worked out how to enable css grid in autoprefixer).

**Comment by jbogarthyde** [comment]
@alan-agius Can you help confirm the current state and help me figure out what to say about it?

File: aio/content/guide/browser-support.md [hunk 1]
```diff
@@ -120,6 +118,15 @@ Note that polyfills cannot magically transform an old, slow browser into a moder
 
 </div>
 
+In Angular CLI version 8 and higher, applications are built using *differential loading*, a strategy where the CLI builds two separate bundles as part of your deployed application.
+
+* The modern bundle contains modern syntax, takes advantage of built-in support in modern browsers, ships less polyfills, and results in a smaller bundle size.
+
+* The second bundle, includes the additional compiled code, all necessary polyfills, and results in a larger bundle size.
```
Comment (alan-agius4) [inline aio/content/guide/browser-support.md]: `The second bundle, includes the additional compiled code, ` -> `The second bundle contains code in ES5 syntax`

File: aio/content/guide/browser-support.md [hunk 2]
```diff
@@ -120,6 +118,15 @@ Note that polyfills cannot magically transform an old, slow browser into a moder
 
 </div>
 
+In Angular CLI version 8 and higher, applications are built using *differential loading*, a strategy where the CLI builds two separate bundles as part of your deployed application.
+
+* The modern bundle contains modern syntax, takes advantage of built-in support in modern browsers, ships less polyfills, and results in a smaller bundle size.
```
Comment (alan-agius4) [inline aio/content/guide/browser-support.md]: `The modern bundle contains modern syntax,` -> The first bundle contains modern ES1015 syntax,

File: aio/content/guide/deployment.md [hunk 3]
```diff
@@ -370,34 +370,58 @@ the subfolder is `my/app/` and you should add `<base href="/my/app/">` to the se
...
+The code you write in development using TypeScript is compiled and bundled into a format that is compatible with most browsers, commonly known as ES5.
+Polyfills are used bridge the gap, providing functionality that simply doesn't exist in some legacy browsers.
 
-There is a cost to ensure this browser compatibility, and it comes in the form of larger bundle size. All modern browsers support ES2015 and beyond, but in most cases, you still have to account for users accessing your application from a browser that doesn't. To maximize compatibility, you ship a single bundle that includes all your compiled code, plus any polyfills that may be needed. Users with modern browsers shouldn't pay the price of increased bundle size when used in a modern browser that supports many of the latest features in JavaScript. This is where differential loading comes into play.
+There is a cost to ensure this browser compatibility, and it comes in the form of larger bundle size.
+All modern browsers support ES2015 and beyond, but in most cases, you still have to account for users accessing your application from a browser that doesn't.
```
Comment (alan-agius4) [inline aio/content/guide/deployment.md]: This paragraph is duplicate of the one in https://github.com/angular/angular/pull/31115/files#diff-b670ec45a1dbc047e58199f72fd57371, 

If we are going to keep this, kindly apply the suggested fixes as well.

File: aio/content/guide/deployment.md [hunk 4]
```diff
@@ -413,25 +437,24 @@ The `index.html` file is also modified during the build process to include scrip
...
+   Some legacy browsers still download both bundles, but only execute the appropriate scripts based on the attributes mentioned above. You can read more on the issue [here](https://github.com/philipwalton/webpack-esnext-boilerplate/issues/1).
 
 </div>
 
 See the [configuration table](#configuration-table) below for the configurations for enabling differential loading.
 
 ### Configuring differential loading
 
-Differential loading for creating builds is already supported with version 8 and later of the Angular CLI. For each application project in your workspace, you can configure how builds are produced based on the mentioned `browserslist` and `tsconfig.json` files in your application project.
+Differential loading for creating builds is supported by default with version 8 and later of the Angular CLI.
```
Comment (alan-agius4) [inline aio/content/guide/deployment.md]: `Differential loading for creating builds is` -> `Differential loading is`

File: aio/content/guide/deployment.md [hunk 5]
```diff
@@ -490,9 +512,69 @@ When the ES5 Browserslist result is `disabled`, then ES5 browser support is not
...
+
+{@ differential-serve}
+
+### Differential loading for test and serve
+
+In Angular version 8 differential loading is enabled by default for the `ng build` command.
+The `ng test` and `ng serve` commands, however, generate a single ES2015 build which cannot run in IE 11.
+
+If you want to have ES5 code during serve, you could disable differential loading completely.
+To maintain the benefits of differential loading, however, a better option is to define multiple configurations for `ng serve` or `ng test`.
```
Comment (alan-agius4) [inline aio/content/guide/deployment.md]: `ng serve` or `ng test`. - >for `ng serve`, `ng e2e` or  `ng test`.

File: aio/content/guide/deployment.md [hunk 6]
```diff
@@ -490,9 +512,69 @@ When the ES5 Browserslist result is `disabled`, then ES5 browser support is not
...
+Differential loading can be explicitly disabled if it causes unexpected issues or you need to target ES5 specifically for legacy browser support.
 
 To explicitly disable differential loading:
 
 - Enable the `dead` or `IE` browsers in the `browserslist` config file by removing the `not` keyword in front of them.
 - Set the `target` in the `compilerOptions` to `es5`.
+
+{@ differential-serve}
+
+### Differential loading for test and serve
```
Comment (alan-agius4) [inline aio/content/guide/deployment.md]: Testing and serving on older browsers

In Angular version 8 differential loading is enabled by default for the `ng build` command.
The `ng test` and `ng serve` commands, however, generate a single ES2015 build which cannot run in older browsers which don't support the modules, such as IE 11.

File: aio/content/guide/deployment.md [hunk 7]
```diff
@@ -490,9 +512,69 @@ When the ES5 Browserslist result is `disabled`, then ES5 browser support is not
...
+
+In Angular version 8 differential loading is enabled by default for the `ng build` command.
+The `ng test` and `ng serve` commands, however, generate a single ES2015 build which cannot run in IE 11.
+
+If you want to have ES5 code during serve, you could disable differential loading completely.
+To maintain the benefits of differential loading, however, a better option is to define multiple configurations for `ng serve` or `ng test`.
+
```
Comment (alan-agius4) [inline aio/content/guide/deployment.md]: Make we add a sub heading `Serve` so that we can link to it.?

File: aio/content/guide/deployment.md [hunk 8]
```diff
@@ -490,9 +512,69 @@ When the ES5 Browserslist result is `disabled`, then ES5 browser support is not
...
+      "browserTarget": "app:build:es5"
+    }
+  }
+},
+</code-example>
+
+You can then run the serve with this configuration.
+
+<code-example language="none" class="code-shell">
+ng serve --configuration es5
```
Comment (alan-agius4) [inline aio/content/guide/deployment.md]: Maybe we should add what needs to be done for e2e and tests?


## Test

Create a new file, `tsconfig-es5.spec.json` next to `tsconfig.spec.json` with the following content.

<code-example language="json" format="." linenums="false">
{
 "extends": "./tsconfig.spec.json",
 "compilerOptions": {
     "target": "es5"
  }
}
</code-example>

<code-example language="json" format="." linenums="false">
"test": {
  "builder": "@angular-devkit/build-angular:karma",
  "options": {
      ...
  },
  "configurations": {
    "es5": {
      "tsConfig": "./tsconfig-es5.spec.json"
    }
  }
},
</code-example>

You can then run the tests with this configuration

<code-example language="none" class="code-shell">
ng test --configuration es5
</code-example>

## E2E

Create an ES5 serve configuration as explained above (link to the above serve section), and configuration an ES5 configuration for the E2E target.
<code-example language="json" format="." linenums="false">
"test": {
  "builder": "@angular-devkit/build-angular:protractor",
  "options": {
      ...
  },
  "configurations": {
	"production": {
		...
	},
    "es5": {
      "devServerTarget": "app:serve:es5"
    }
  }
},
</code-example>

You can then run the e2e's with this configuration
<code-example language="none" class="code-shell">
ng e2e --configuration es5
</code-example>

File: aio/content/guide/deployment.md (hunk 6 above)
Comment (IgorMinar) [reply aio/content/guide/deployment.md]: I was going to suggest something similar as well. "Testing and serving on older browsers" reflects what the users are doing better than the actual commands.

consider making the title even more explicit: "Testing and local development with older browsers"

**Review by IgorMinar — APPROVED** [review]
the rest looks good after Alan's comments are resolved. thank you!

**Comment by brandonroberts** [comment]
Nit: The commit messages say `e5` target instead of `es5` target

**Comment by brandonroberts** [comment]
@alastair-todd agreed that the reflect metadata polyfills are no longer required and should be removed from the polyfills page. I think it should be done in a separate PR to make sure there aren't other references that need to be removed. Will you file a separate issue for that and about mentioning autoprefixer?

File: aio/content/guide/deployment.md [hunk 9]
```diff
@@ -490,9 +539,151 @@ When the ES5 Browserslist result is `disabled`, then ES5 browser support is not
...
+
+## Local development in older browsers
+
+In Angular CLI version 8 and higher, differential loading is enabled by default for the `ng build` command.
+The `ng serve`, `ng test`, and `ng e2e` commands, however, generate a single ES2015 build which cannot run in older browsers that don't support the modules, such as IE 11.
+
+If you want to run ES5 code during development, you could disable differential loading completely.
+To maintain the benefits of differential loading, however, a better option is to define multiple configurations for `ng serve`, `ng e2e`, and `ng test`.
+
+{@ differential-serve}
```
Comment (brandonroberts) [inline aio/content/guide/deployment.md]:
```suggestion
{@a differential-serve}
```

Comment (jbogarthyde) [reply aio/content/guide/deployment.md]: Thanks! this was driving me crazy!

**Pull request closed without merging**